sinks/influxdb/driver.go

```diff
@@ -85,101 +94,83 @@ func (self *influxdbSink) metricToSeries(timeseries *sink_api.Timeseries) *influ
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
 
-	return self.newSeries(seriesName, columns, values)
-}
-
-var eventColumns = []string{
-	"time",                     // Column 0
-	"sequence_number",          // Column 1
-	sink_api.LabelPodId.Key,    // Column 2
-	sink_api.LabelPodName.Key,  // Column 3
-	sink_api.LabelHostname.Key, // Column 4
-	"value",                    // Column 5
+	return point
 }
 
 // Stores events into the backend.
 func (sink *influxdbSink) StoreEvents(events []kube_api.Event) error {
-	dataPoints := []*influxdb.Series{}
+	if err := sink.createDatabase(); err != nil {
+		return err
+	}
 	if events == nil || len(events) <= 0 {
 		return nil
 	}
-	if !sink.c.avoidColumns {
-		dataPoint, err := sink.storeEventsColumns(events)
-		if err != nil {
-			glog.Errorf("failed to parse events: %v", err)
-			return err
-		}
-		dataPoints = append(dataPoints, dataPoint)
-	} else {
-		for _, event := range events {
-			dataPoint, err := sink.storeEventNoColumns(event)
-			if err != nil {
-				glog.Errorf("failed to parse events: %v", err)
-				return err
-			}
-			dataPoints = append(dataPoints, dataPoint)
-		}
-	}
-	err := sink.client.WriteSeriesWithTimePrecision(dataPoints, influxdb.Millisecond)
+	points, err := sink.eventsToPoints(events)
 	if err != nil {
+		glog.Errorf("failed to parse events: %v", err)
+		return err
+	}
+	bp := influxdb.BatchPoints{
+		Points:   points,
+		Database: sink.c.dbName,
+	}
+	if _, err = sink.client.Write(bp); err != nil {
+		if strings.Contains(err.Error(), dbNotFoundError) {
+			sink.dbExists = false
+		}
 		glog.Errorf("failed to write events to influxDB - %s", err)
 		sink.recordWriteFailure()
 	} else {
-		glog.V(1).Info("Successfully flushed events to influxDB")
+		glog.V(4).Info("Successfully flushed events to influxDB")
 	}
-	return err
 
+	return err
 }
 
-func (sink *influxdbSink) storeEventsColumns(events []kube_api.Event) (*influxdb.Series, error) {
+func (sink *influxdbSink) eventsToPoints(events []kube_api.Event) ([]influxdb.Point, error) {
 	if events == nil || len(events) <= 0 {
 		return nil, nil
 	}
-	points := make([][]interface{}, len(events))
-	for i, event := range events {
-		points[i] = make([]interface{}, len(eventColumns))
-		points[i][0] = event.LastTimestamp.Time.UTC().Round(time.Millisecond).Unix() // Column 0 - time
-		points[i][1] = hashUID(string(event.UID))                                    // Column 1 - sequence_number
-		if event.InvolvedObject.Kind == "Pod" {
-			points[i][2] = event.InvolvedObject.UID  // Column 2 - pod_id
-			points[i][3] = event.InvolvedObject.Name // Column 3 - pod_name
-		} else {
-			points[i][2] = "" // Column 2 - pod_id
-			points[i][3] = "" // Column 3 - pod_name
-		}
+	points := make(influxdb.Point, 0, len(events))
```

Comment:
I could not compile any of the code because of missing godeps. If this PR looks good in general I can add godeps again.